examples/domainmodel/test/cross-refs.test.ts

```diff
@@ -0,0 +1,58 @@
+/******************************************************************************
+ * Copyright 2021 TypeFox GmbH
+ * This program and the accompanying materials are made available under the
+ * terms of the MIT License, which is available in the project root.
+ ******************************************************************************/
+
+import { AstNode, LangiumDocument, ReferenceDescription } from "langium";
+import { DefaultIndexManager } from "langium/lib/index/index-manager";
+import { parseDocument } from "langium/lib/test";
+import { createDomainModelServices } from "../src/language-server/domain-model-module";
+import { Domainmodel } from "../src/language-server/generated/ast";
+
+const services = createDomainModelServices()
+
+const datatypeFile = `
+datatype String
+datatype Int
+datatype Decimal
+
+package big {
+    datatype Int
+    datatype Decimal
+}
+`;
+
+const referencingFile = `
+ entity Blog {
+    title: String
+    description: String
+    time: big.Int
+}
+`;
+describe('Cross references', () => {
+    const datatypeDoc: LangiumDocument = parseDocument(services, datatypeFile);
+    const referencingDoc: LangiumDocument = parseDocument(services, referencingFile);
+    (services.index.IndexManager as DefaultIndexManager).processDocuments([referencingDoc, datatypeDoc]);
```

Comment:
Shall we expose that method in the interface? Type-casting services always looks like a bad smell to me.

Reply:
This is a test class, I'm not yet sure wie will need this somewhere in the real projects, but yes why not.

Reply:
We should add documentation to the IndexManager methods about when and how to use them.